### ISSUE TITLE
nowarn_deprecated_function doesn't apply to callbacks

### ISSUE THREAD

**Issue by mkuratczyk** [erlang/otp#8331]
OTP 27 deprecates `gen_server:format_status/2`. I wanted to ignore this warning temporarily. However, it seems like `nowarn_deprecated_function` attribute doesn't apply to callbacks:

Given `test.erl` like this:
```
-module(test).

-compile(nowarn_deprecated_function).
-compile({nowarn_deprecated_function, {gen_server, format_status, 2}}).

-behaviour(gen_server).

-export([format_status/2]).
-export([init/1, handle_call/3, handle_cast/2]).

init(_) ->
    {ok, {}}.

handle_call(_, _, _) ->
    {noreply, {}}.

handle_cast(_, _) ->
    {noreply, {}}.

format_status(_, _) ->
    {}.

```

I still get the warning:
```
> erlc test.erl
test.erl:6:2: Warning: the callback gen_server:format_status(_,_) is deprecated; use format_status/1 instead
%    6| -behaviour(gen_server).
%     |  ^
```

If there's a way to disable this warning, I'd appreciate your help.

**Affected versions**
OTP27-rc.2 (likely all previous versions as well).

**Comment by garazdawi** [comment]
You should be able to use `-compile(nowarn_deprecated_callback).`. Can you try that?

**Comment by mkuratczyk** [comment]
Yes, this does the trick, thank you! As far as I can tell it's undocumented but works. :)

**Comment by garazdawi** [comment]
> As far as I can tell it's undocumented but works. :)

Yes, I noticed as much when looking for it. Not sure how I managed to miss that when implementing it. Fixed in #8332.
